test/functional-test/functional-test-lib/taskTestRunner.ts

```diff
@@ -0,0 +1,80 @@
+import * as cp from 'child_process';
+import path = require('path');
+import process = require('process');
+
+export interface TaskInfo {
+  name: string;
+  path: string;
+  inputVariables?: EnvironmentVariableDefinition[];
+}
+
+export interface TaskResult {
+  processResult:cp.SpawnSyncReturns<string>;
+  outputEnvironmentVariable?: EnvironmentVariableDefinition;
+}
+
+export interface EnvironmentVariableDefinition {
+  name: string;
+  value: string;
+}
+
+export class TaskRunner {
+  private taskResult:cp.SpawnSyncReturns<string> | undefined = undefined;
+  taskInfo: TaskInfo;
+  taskDirectory: string;
+
+  constructor(taskInfo: TaskInfo, taskDirectory: string) {
+    this.taskInfo = taskInfo;
+    this.taskDirectory = taskDirectory;
+  }
+
+  runTask(): TaskResult{
+    const normalizedTaskPath = this.normalizeAbsoluteTaskPath()
+    this.taskResult = cp.spawnSync('node', [normalizedTaskPath], { encoding: 'utf-8', cwd: this.taskDirectory });
+    console.debug(this.taskResult.stdout);
+    this.validateTaskRun();
+    var envVar = this.setOutputEnvironmentVariables();
+    return {processResult: this.taskResult, outputEnvironmentVariable: envVar };
+  }
+
+  private normalizeAbsoluteTaskPath() {
+    return (path.join(this.taskDirectory, this.taskInfo.path)).replace(/\\/g, '/');
+  }
+
+  private setOutputEnvironmentVariables(): EnvironmentVariableDefinition | undefined {
+    if(!this.taskResult) return;
```

Comment:
Style nit: space between if and parenthesis
`if (!this.taskResult) return;`
ditto below in line 55